tools/reviewer/job/sync/DiffConverter.java

```diff
@@ -0,0 +1,178 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.reviewer.job.sync;
+
+import com.google.common.collect.ImmutableList;
+import com.google.startupos.tools.reviewer.job.sync.GithubProtos.CommitRequest;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.IssueComment;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.PullRequest;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.ReviewComment;
+import com.google.startupos.tools.reviewer.job.sync.GithubPullRequestProtos.User;
+import com.google.startupos.tools.reviewer.localserver.service.Protos;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Comment;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Diff;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.GithubPr;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Thread;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+// Converts a Diff proto to a list of GithubPullRequestProtos.PullRequest protos
+public class DiffConverter {
+
+  private GithubClient githubClient;
+
+  public DiffConverter(GithubClient githubClient) {
+    this.githubClient = githubClient;
+  }
+
+  public List<PullRequest> toPullRequests(Diff diff) {
+    List<PullRequest> pullRequests = new ArrayList<>();
+
+    for (GithubPr githubPr : diff.getGithubPrList()) {
+      PullRequest.Builder pullRequest = PullRequest.newBuilder();
+
+      if (githubPr.getNumber() != 0) {
+        pullRequest.setNumber(githubPr.getNumber());
+      }
+      pullRequest
+          .setState(getPullRequestState(diff.getStatus()))
+          .setTitle("D" + diff.getId())
+          .setUser(User.newBuilder().setEmail(diff.getAuthor().getEmail()).build())
+          .setBody(diff.getDescription())
+          .setCreatedAt(Instant.ofEpochMilli(diff.getCreatedTimestamp()).toString())
+          .setUpdatedAt(Instant.ofEpochMilli(diff.getModifiedTimestamp()).toString())
+          .setBaseBranchName("master")
+          .setHeadBranchName("D" + diff.getId())
+          .setRepo(githubPr.getRepo())
+          .addAllReviewComment(getReviewCommentsByRepoName(diff, githubPr))
+          .addAllIssueComment(getIssueComments(diff, githubPr))
+          .setOwner(githubPr.getOwner())
+          .setAssociatedReviewerDiff(diff.getId());
+      pullRequests.add(pullRequest.build());
+    }
+    return pullRequests;
+  }
+
+  private ImmutableList<ReviewComment> getReviewCommentsByRepoName(Diff diff, GithubPr githubPr) {
```

Comment:
I think, either rename `s/getReviewCommentsByRepoName/getReviewCommentsByGithubPr`, or pass `String repoName` (=`githubPr.getRepo()`) instead of `GithubPr githubPr`.

Reply:
Done.